src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/CompleteNextDataSetVersionImportFunction.cs

```diff
@@ -0,0 +1,177 @@
+            .Imports
+            .SingleOrDefault(import => import.DataSetVersionId == nextDataSetVersion.Id
+                                       && import.Stage == DataSetVersionImportStage.ManualMapping);
+
+        return importToContinue is null
+            ? CreateDataSetVersionIdError(
+                message: ValidationMessages.ImportInManualMappingStateNotFound,
+                dataSetVersionId: request.DataSetVersionId)
+            : importToContinue;
+    }
+
+    private async Task<Either<ActionResult, DataSetVersion>> GetNextDataSetVersionInMappingStatus(
+        DataSetVersionProcessRequest request,
+        CancellationToken cancellationToken)
+    {
+        var nextVersion = await publicDataDbContext
+            .DataSetVersions
+            .AsNoTracking()
+            .Include(dataSetVersion => dataSetVersion.Imports)
+            .SingleOrDefaultAsync(
+                dataSetVersion => dataSetVersion.Id == request.DataSetVersionId,
+                cancellationToken);
+
+        if (nextVersion is null)
+        {
+            return CreateDataSetVersionIdError(
+                message: ValidationMessages.DataSetVersionNotFound,
+                dataSetVersionId: request.DataSetVersionId);
+        }
+
+        if (nextVersion.Status != DataSetVersionStatus.Mapping)
+        {
+            return CreateDataSetVersionIdError(
+                message: ValidationMessages.DataSetVersionNotInMappingStatus,
+                dataSetVersionId: request.DataSetVersionId);
+        }
+
+        return nextVersion;
+    }
+
+    private async Task<Either<ActionResult, DataSetVersionMapping>> GetCompletedDataSetVersionMapping(
+        DataSetVersionProcessRequest request,
+        CancellationToken cancellationToken)
+    {
+        var mapping = await publicDataDbContext
+            .DataSetVersionMappings
+            .AsNoTracking()
+            .SingleOrDefaultAsync(
+                mapping => mapping.TargetDataSetVersionId == request.DataSetVersionId,
+                cancellationToken);
+
+        if (mapping is null)
+        {
+            return CreateDataSetVersionIdError(
+                message: ValidationMessages.DataSetVersionMappingNotFound,
+                dataSetVersionId: request.DataSetVersionId);
+        }
+
+        if (!mapping.FilterMappingsComplete || !mapping.LocationMappingsComplete)
+        {
+            return CreateDataSetVersionIdError(
+                message: ValidationMessages.DataSetVersionMappingsNotComplete,
+                dataSetVersionId: request.DataSetVersionId);
+        }
+
+        return mapping;
+    }
+
+    private async Task ProcessCompletionOfNextDataSetVersionImport(
+        DurableTaskClient client,
+        Guid dataSetVersionId,
+        Guid instanceId,
+        CancellationToken cancellationToken)
+    {
+        const string orchestratorName =
+            nameof(ProcessCompletionOfNextDataSetVersionFunction.CompleteNextDataSetVersionImportProcessing);
```

Comment:
should this be `nameof(ProcessCompletionOfNextDataSetVersionFunction.ProcessCompletionOfNextDataSetVersion)`?

Reply:
or are we waiting until 4954 to switch it out?

Reply:
Nope you're right, it's the incorrect name here.  Changed, thanks